### PR TITLE
[OpenVINO] LFM2-MoE support for transformers v4.

### PR DESCRIPTION
**Pull request by popovaan**
# What does this PR do?

Added LFM2-MoE support for transformers v4.

## Before submitting
- [N/A] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [N/A] Did you make sure to update the documentation with your changes?
- [N/A as the model is private.] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/exporters/openvino/model_configs.py [hunk 1]
```diff
@@ -4902,6 +4582,22 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
...
 
+@register_in_tasks_manager(
+    "lfm2_moe",
+    *[
+        "text-generation",
+        "text-generation-with-past",
+    ],
+    library_name="transformers",
+)
+class LFM2OpenVINOConfig(LFM2OpenVINOConfig):
```
Comment (rkazants) [inline optimum/exporters/openvino/model_configs.py]:
```suggestion
class LFM2MoeOpenVINOConfig(LFM2OpenVINOConfig):
```

File: optimum/exporters/openvino/model_configs.py [hunk 2]
```diff
@@ -4902,6 +4582,22 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
...
+    MIN_TRANSFORMERS_VERSION = "4.54.0"
```
Comment (rkazants) [inline optimum/exporters/openvino/model_configs.py]: let us put `MAX_TRANSFORMERS_VERSION = 4.57.99`

File: optimum/exporters/openvino/model_patcher.py [hunk 3]
```diff
@@ -7280,12 +7167,101 @@ def __exit__(self, exc_type, exc_value, traceback):
...
+        num_experts,
+        device=hidden_states.device,
+        dtype=hidden_states.dtype,
+    )
+    dense_routing_weights.scatter_(dim=1, index=selected_experts, src=routing_weights)
+    hidden_states_expanded = hidden_states.repeat(num_experts, 1) # (num_experts * num_tokens, hidden_dim)
+    hidden_states_expanded = hidden_states_expanded.view(num_experts, -1, hidden_dim) # (num_experts, num_tokens, hidden_dim)
+
+    # Stack expert parameters
+    w1_stacked = torch.stack([e.w1.weight.T for e in self.experts])
```
Comment (rkazants) [inline optimum/exporters/openvino/model_patcher.py]: you also transpose weight, right? May be not needed

Comment (popovaan) [reply optimum/exporters/openvino/model_patcher.py]: No, without it shapes don't match. 
I see that in Afmoe these weights are transposed as well, but concatenated beforehand. I can implement it the same way https://github.com/huggingface/optimum-intel/blob/2c48d6430c265ac259c1b264f3e2c4025cdd7b76/optimum/exporters/openvino/model_patcher.py#L7604C16-L7611C18

File: optimum/exporters/openvino/model_configs.py (hunk 2 above)
Comment (popovaan) [reply optimum/exporters/openvino/model_configs.py]: Done.

File: optimum/exporters/openvino/model_configs.py [hunk 4]
```diff
@@ -163,6 +163,7 @@
     InternVLChatImageEmbeddingModelPatcher,
     JaisModelPatcher,
```
Comment (rkazants) [inline optimum/exporters/openvino/model_configs.py:164]: please update https://github.com/huggingface/optimum-intel/blob/main/optimum/exporters/openvino/utils.py#L308 with `lfm2_moe`. It can help with an inference

File: optimum/exporters/openvino/model_patcher.py (hunk 3 above)
Comment (rkazants) [reply optimum/exporters/openvino/model_patcher.py]: Got it

Comment (rkazants) [reply optimum/exporters/openvino/model_patcher.py]: @popovaan, btw, can we re-use existing MoE patching for this model?

File: optimum/exporters/openvino/model_configs.py (hunk 4 above)
Comment (popovaan) [reply optimum/exporters/openvino/model_configs.py:164]: Yes, it helped, now the model can be infered and text looks adequate, but it doesn't match original model text.

File: optimum/exporters/openvino/model_patcher.py (hunk 3 above)
Comment (popovaan) [reply optimum/exporters/openvino/model_patcher.py]: The closest patching to this version of MoE is Qwen3 MoE block, but it has a slightly diffrent preprocessing of routing_weights (sofmax is applied instead of dividing by sum), so I suppose we can't reuse it as is.

File: optimum/exporters/openvino/model_configs.py (hunk 4 above)
Comment (popovaan) [reply optimum/exporters/openvino/model_configs.py:164]: Changed the patching in this commit, now the output text seems fine: https://github.com/huggingface/optimum-intel/pull/1609/commits/e7ad8fb036818dd9994e9a65e425148dcb399b15

Comment (popovaan) [reply optimum/exporters/openvino/model_configs.py:164]: Returned torch.bmm patching as agreed. Accuracy is fine with it.

File: optimum/exporters/openvino/model_configs.py [hunk 5]
```diff
@@ -4902,6 +4582,22 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
...
+        "text-generation-with-past",
+    ],
+    library_name="transformers",
+)
+class LFM2MoeOpenVINOConfig(LFM2OpenVINOConfig):
+    MAX_TRANSFORMERS_VERSION = "4.57.99"
+    _MODEL_PATCHER = Lfm2MoeModelPatcher
+
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator, Lfm2DummyPastKeyValuesGenerator)
+    DUMMY_PKV_GENERATOR_CLASS = Lfm2DummyPastKeyValuesGenerator
```
Comment (rkazants) [inline optimum/exporters/openvino/model_configs.py]:
```suggestion
class LFM2MoeOpenVINOConfig(LFM2OpenVINOConfig):
    MAX_TRANSFORMERS_VERSION = "4.57.99"
    _MODEL_PATCHER = Lfm2MoeModelPatcher
```

File: optimum/exporters/openvino/model_patcher.py [hunk 6]
```diff
@@ -7275,17 +7162,110 @@ def __enter__(self):
...
+                    tuple(sparse_moe_block.experts[i].w1.weight.unsqueeze(0) for i in range(num_experts)),
+                    dim=0,
+                ).transpose(1, 2)
+                sparse_moe_block.w2_stacked = torch.concat(
+                    tuple(sparse_moe_block.experts[i].w2.weight.unsqueeze(0) for i in range(num_experts)),
+                    dim=0,
+                ).transpose(1, 2)
+                sparse_moe_block.w3_stacked = torch.concat(
+                    tuple(sparse_moe_block.experts[i].w3.weight.unsqueeze(0) for i in range(num_experts)), dim=0
+                ).transpose(1, 2)
```
Comment (rkazants) [inline optimum/exporters/openvino/model_patcher.py]: please remove weight constants in `exit`

Comment (popovaan) [reply optimum/exporters/openvino/model_patcher.py]: Done.

**Comment by rkazants** [comment]
Already merged another PR for official LFM2-MoE
